tools/i18n-plugin/src/main/java/org/openhab/core/tools/i18n/plugin/XmlToTranslationsConverter.java

```diff
@@ -0,0 +1,368 @@
+                            .add(entry(String.format("%s.label", groupTypeKeyPrefix), channelGroupType.getLabel()));
+
+                    entriesBuilder.add(entry(String.format("%s.description", groupTypeKeyPrefix),
+                            channelGroupType.getDescription()));
+
+                    channelGroupType.getChannelDefinitions().stream() //
+                            .sorted(Comparator.comparing(ChannelDefinition::getId)) //
+                            .forEach(channelDefinition -> {
+                                String label = channelDefinition.getLabel();
+                                if (label != null) {
+                                    entriesBuilder.add(entry(String.format("%s.channel.%s.label", groupTypeKeyPrefix,
+                                            channelDefinition.getId()), label));
+                                }
+
+                                String description = channelDefinition.getDescription();
+                                if (description != null) {
+                                    entriesBuilder.add(entry(String.format("%s.channel.%s.description",
+                                            groupTypeKeyPrefix, channelDefinition.getId()), description));
+                                }
+                            });
+
+                    return group(entriesBuilder.build());
+                }));
+    }
+
+    private TranslationsSection channelTypesSection(BundleInfo bundleInfo) {
+        String header = "channel types";
+        List<ChannelTypeXmlResult> channelTypesXml = bundleInfo.getChannelTypesXml();
+        if (channelTypesXml.isEmpty()) {
+            return section(header);
+        }
+
+        String keyPrefix = String.format("channel-type.%s", bundleInfo.getBindingId());
+
+        return section(header, channelTypesXml.stream().map(channelTypeXml -> {
+            ChannelType channelType = channelTypeXml.toChannelType();
+            String channelId = channelType.getUID().getId();
+
+            Builder<TranslationsEntry> entriesBuilder = Stream.builder();
+            entriesBuilder.add(entry(String.format("%s.%s.label", keyPrefix, channelId), channelType.getLabel()));
+            entriesBuilder
+                    .add(entry(String.format("%s.%s.description", keyPrefix, channelId), channelType.getDescription()));
+
+            StateDescription stateDescription = channelType.getState();
+            if (stateDescription != null) {
+                stateDescription.getOptions().stream()
+                        .map(option -> entry(
+                                String.format("%s.%s.state.option.%s", keyPrefix, channelId, option.getValue()),
+                                option.getLabel()))
+                        .forEach(entriesBuilder::add);
+
+                if (stateDescription.getPattern() != null) {
+                    String pattern = stateDescription.getPattern();
+                    if (pattern != null && pattern.contains("%1$t")) {
```

Comment:
Did you add this check (`pattern.contains("%1$t")`) by intention to filter every other pattern except date time related ones? I was looking into this because I am missing pattern for channel types from feed binding for translation. It uses `"%tc"` as pattern. I personally would be fine with this and change it for feed binding to something else. Or should we extend it here?

Reply:
See https://github.com/openhab/openhab-addons/pull/11618

Reply:
I added the filter because when it is not applied it will also start adding the patterns for the following states to .properties files:

* `<state pattern="%s">`
* `<state pattern="%.2f" readOnly="true"/>`
* `<state pattern="%.1f %unit%"/>`

I doubt these would often result in other translations.

Although later I realised that any string that has a space in it, is also a potential translation candidate for any language that is right-to-left instead of left to right. :slightly_smiling_face: